File: POO JAVA/Aula07/src/aula07/Luta.java
package aula07;

import java.util.Random;

public class Luta {

    //Atributos
    private Lutador desafiado;
    private Lutador desafiante;
    private int rounds;
    private boolean aprovada;
    
    
    //Metodos Acessores e Modificadores
    private Lutador getDesafiado() {
        return desafiado;
    }
    private void setDesafiado(Lutador desafiado) {
        this.desafiado = desafiado;
    }
    private Lutador getDesafiante() {
        return desafiante;
    }
    private void setDesafiante(Lutador desafiante) {
        this.desafiante = desafiante;
    }
    private int getRounds() {
        return rounds;
    }
    private void setRounds(int rounds) {
        this.rounds = rounds;
    }
    private boolean getAprovada() {
        return aprovada;
    }
    private void setAprovada(boolean aprovada) {
        this.aprovada = aprovada;
    }
    
    //Metodos da Classe
    public void marcarLuta(Lutador m1, Lutador m2){
        if((m1.getCategoria().equals(m2.getCategoria())) && (m1 != m2)){
            this.aprovada = true;
            this.desafiado = m1;
            this.desafiante = m2;
        }else{
            this.aprovada = false;
            this.desafiado = null;
            this.desafiante = null;
        }
    }
    public void lutar(){
        if(this.getAprovada()){
            System.out.println("");
            System.out.println("==== Desafiante ====");
            this.getDesafiado().apresentar();
            System.out.println("");
            System.out.println("==== Desafiado ====");
            this.getDesafiante().apresentar();
            
            Random aleatorio = new Random();
            int vencedor = aleatorio.nextInt(3);
            System.out.println("");
            System.out.println("==== RESULTADO DA LUTA ====");
            
            switch(vencedor){
                case 0: //Empatar Luta
                    System.out.println("Lutar ficou Empatada !!!");
                    this.desafiado.empatarLuta();
                    this.desafiante.empatarLuta();
                    break;
                case 1: //Vencedor desafiado
                    System.out.println("Vitoria do "+this.desafiado.getNome());
                    this.desafiado.ganharLuta();
                    this.desafiante.perderLuta();
                    break;
                case 2://Vencedor desafiante
                    System.out.println("Vitoria do "+this.desafiante.getNome());
                    this.desafiado.perderLuta();
                    this.desafiante.ganharLuta();
                    break;
            }
            
        }else{
            System.out.println("");
            System.out.println("==============================");
            System.out.println("A lutar nao pode acontecer.");
        }
        System.out.println("");
    }
    
}

File: POO PHP/Aula07/Luta.php
<?php
require_once './Lutador.php';
class Luta {

    //Atributos da Classe
    private $desafiado;
    private $desafiante;
    private $rounds;
    private $aprovada;
    
    
    //Metodos Acessores e Modificadores
    public function getDesafiado(){
        return $this->desafiado;
    }
    public function getDesafiante(){
        return $this->desafiante;
    }
    public function getRounds(){
        return $this->rounds;
    }
    public function getAprovada(){
        return $this->aprovada;
    }
    public function setDesafiado($desafiado){
        $this->desafiado = $desafiado;
    }
    public function setDesafiante($desafiante){
        $this->desafiante = $desafiante;
    }
    public function setRounds($rounds){
        $this->rounds = $rounds;
    }
    public function setAprovada($aprovada){
        $this->aprovada = $aprovada;
    }

    //Metodos da Classe
    public function marcarLuta($L1, $L2){
        if(($L1->getCategoria() === $L2->getCategoria()) && ($L1 != $L2)){
            $this->aprovada = true;
            $this->desafiado = $L1;
            $this->desafiante = $L2;
        } else {
            $this->aprovada = false;
            $this->desafiado = null;
            $this->desafiante = null;
        }
    }
    public function lutar(){
        if($this->aprovada){
            $this->desafiado->apresentar();
            $this->desafiante->apresentar();
            $vencedor = rand(0,2);
            switch ($vencedor){
                case 0://empate
                    echo "<p>Empate!!!</p>";
                    $this->desafiado->empatarLuta();
                    $this->desafiante->empatarLuta();
                    break;
                case 1://desafiado vence
                    echo "<p>".$this->desafiado->getNome()." venceu.</p>";
                    $this->desafiado->ganharLuta();
                    $this->desafiante->perderLuta();
                    break;
                case 2://desafoante vence
                    echo "<p>".$this->desafiante->getNome()." venceu.</p>";
                    $this->desafiado->perderLuta();
                    $this->desafiante->ganharLuta();
                    break;
            }
        } else {
            echo "<p>A luta não pode acontecer !!!</p>";
        }
    }
}
